music_assistant/common/helpers/uri.py

```diff
@@ -45,6 +61,10 @@ def parse_uri(uri: str) -> tuple[MediaType, str, str]:
     except (TypeError, AttributeError, ValueError, KeyError) as err:
         msg = f"Not a valid Music Assistant uri: {uri}"
         raise MusicAssistantError(msg) from err
```

Comment:
maybe raise InvalidProviderURI here as well ? That way we dont have to catch the very generic MusicAssistantError anymore

Reply:
I think we still need two different named exceptions to catch from `parse_uri`:
1: This is not a URI: pass through to the regular search block
2: This is a URI but it's an invalid provider ID: log and return empty SearchResults() so no search is performed

They could be better named though.  How about:
1: Use InvalidProviderURI
2: Use InvalidProviderID

Reply:
Yes, that sounds good!